ground/uavobjgenerator/uavobjectparser.cpp

```diff
@@ -404,6 +404,12 @@ QString UAVObjectParser::parseXML(QString& xml, QString& filename)
         // Calculate size
         calculateSize(info);
 
+        // Buffer on flight side is 256 bytes
+        // Additional overhead is 2+4+2+2+1+1=12
+        if(info->numBytes > 244)

```

Comment:
I think the issue may be that this module doesn't have visibility to the uavtalk headers and plumbing it may be unfortunate.

I suggest 

```
#define UAVTALK_OVERHEAD (2+4+2+2+1+1)
#define MAX_UAVOSIZE (UINT8_MAX - UAVTALK_OVERHEAD)
```

Later that define can be factored out to uavtalk and replaced with constants for the specific sizes (object ids, lengths, etc).